File: Frontend/src/types/OutgoingDocumentDTO.ts
import { ClientDTO } from "./ClientDTO";
import { OutgoingStateEnum } from "./OutgoingStateEnum";
import { OutgoingResourceDTO } from "./OutgoingResourceDTO";

export interface OutgoingDocumentDTO {
  id: string | null;                 
  number: string;
  clientDTO: ClientDTO;
  date: string;                     
  state: OutgoingStateEnum;
  outgoingResourcesDTO: OutgoingResourceDTO[];
}

File: Backend/TestTask.Application/DTOs/Warehouse/OutgoingDocumentDTO.cs
using TestTask.Application.DTOs.Directories;
using TestTask.Domain.Enums;

namespace TestTask.Application.DTOs.Warehouse
{
    public class OutgoingDocumentDTO
    {
        public Guid? Id { get; set; }
        public string Number { get; set; }
        public ClientDTO ClientDTO { get; set; } = new ClientDTO();
        public DateTime Date { get; set; }
        public DirectoriesStateEnum State { get; set; }
        public List<OutgoingResourceDTO> OutgoingResourcesDTO { get; set; } = new List<OutgoingResourceDTO>();
    }
}
